v2/emailpassword/troubleshooting/cors-issues.mdx

```diff
@@ -0,0 +1,261 @@
+
+```tsx
+import SuperTokens from "supertokens-node";
+import express from "express";
+import cors from "cors";
+let app = express();
+
+// ...other middlewares
+
+app.use(
+  cors({
+    origin: "http://127.0.0.1:8080",
+    allowedHeaders: ["content-type", ...SuperTokens.getAllCORSHeaders()],
+    methods: ["GET", "PUT", "POST", "DELETE"],
+    credentials: true,
+  })  
+);
+
+// ...your API routes
+```
+
+## Different parts of CORS
+
+Let’s take a step back and talk about what each of these fields mean.
+
+**origin [must have]** -
+
+This configures the Access-Control-Allow-Origin CORS header. It expects a couple input types. For our case, we’ll only talk about what happens when we assign a Boolean or String to this field.
+
+- Boolean - set origin to true to reflect the request origin, as defined by req.header('Origin'), or set it to false to disable CORS.
+- String - set origin to a specific origin. For example, once set to "http://example.com" only requests from "http://example.com" will be allowed.
+
+This is important because Supertokens has both a frontend and backend SDK. The frontend SDK calls the APIs exposed by the backend SDK. As a result, if the origin to set to false or the wrong origin, we’ll find an error such as the following -
+
+:::danger
+Origin [http://localhost:3000](http://localhost:3000/) is not allowed by Access-Control-Allow-Origin. Status code: 204
+:::
+
+**allowedHeaders [optional] -**
+
+This configures the Access-Control-Allow-Headers CORS header. It expects one of two input types.
+
+- Comma-delimited string - 'X-Custom-Header, Content-Type'
+- Array - ['X-Custom-Header', 'Content-Type']
+
+For Supertokens projects, we can use the SuperTokens.getCORSAllowedHeaders() function which calls the following list - ["anti-csrf", "rid"] for the Content-Type.
+
+This is important because the header tells the browser what HTTP headers are allowed to be used when making a cross-origin request. By default, CORS will reflect the headers specified in the request's Access-Control-Request-Headers header.
+
+**methods [optional] -**
+
+This configures the Access-Control-Allow-Methods CORS header. It expects one of two input types.
+
+- Comma-delimited string - 'GET,PUT,POST'
+- Array - ['GET', 'PUT', 'POST']
+
+This is important because the header tells the browser what HTTP verbs are allowed to be used when making a cross-origin request. Supertokens in particular uses GETs and POSTs to verify, create new accounts, sign in/out and many other functions.
+
+**credentials [must have] -**
+
+This configures the Access-Control-Allow-Credentials CORS header. It expects a boolean.
+
+If you set this header to true, then the browser will allow cookies to be sent in cross-origin requests. This is important because cookies are the only way Supertokens can store the user's session. It is required to set this header to true, so that the browser can send cookies to the Supertokens service.
```

Comment:
```suggestion
If you set this header to true, then the browser will allow cookies to be sent in cross-origin requests. This is important because SuperTokens, by default uses cookies for sessions in the browser.
```